pyscal/gasoil.py

```diff
@@ -851,28 +842,10 @@ def slgof_df(self) -> pd.DataFrame:
             .reset_index(drop=True)
         )
         # It is a strict requirement that the first sl value should be swl + sorg,
-        # so we modify it if it close. If it is not close, we do not dare to fix
-        # it, to ensure we don't cover bugs.
         slgof_sl_mismatch = abs(slgof["SL"].values[0] - (self.sorg + self.swl))
-        if slgof_sl_mismatch > epsilon:
-            if slgof_sl_mismatch < 2 * 1.0 / float(SWINTEGERS):
-                # Repair the table in-place:
-                slgof.loc[0, "SL"] = self.sorg + self.swl
-                # After modification, we can get duplicate sl values,
-                # so drop duplicates:
-                slgof["slint"] = list(
-                    map(int, list(map(round, slgof["SL"] * SWINTEGERS)))
-                )
-                slgof.drop_duplicates("slint", inplace=True)
-                # Delete the temporary column:
-                slgof.drop(labels="slint", axis="columns", inplace=True)
-            else:
-                # Give up repairing the table:
-                logger.critical(
-                    "SLGOF does not start at the correct value. Please report as bug."
-                )
-                logger.error("slgof_sl_mismatch: %f", slgof_sl_mismatch)
-                logger.error(str(slgof.head()))
+        assert (
+            slgof_sl_mismatch < epsilon
+        ), f"Bug: slgof_sl_mismatch was {slgof_sl_mismatch}, too large"
```

Comment:
Same here, really. You can run `python -O -m pyscal` and ignore all `assert`. Is that what you want?

Reply:
Changed to `if` and `ValueError`.